### PR TITLE
[CONTINT-4239] Fix namespace label as tags config and error log

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR fixes the config to enable the namespace informer in the kubeapiserver collector.
Also, the error log was spammed because of the misconfiguration.
```
2024-06-04 11:54:54 UTC | CLUSTER | ERROR | (api/v1/kubernetes_metadata.go:121 in ]) | Could not retrieve the local-path-storage of namespace metadata: "kubernetes_namespace" not found
2024-06-04 11:54:54 UTC | CLUSTER | ERROR | (api/v1/kubernetes_metadata.go:121 in ]) | Could not retrieve the datadog-agent-helm of namespace metadata: "kubernetes_namespace" not found
2024-06-04 11:54:54 UTC | CLUSTER | ERROR | (api/v1/kubernetes_metadata.go:121 in ]) | Could not retrieve the kube-system of namespace metadata: "kubernetes_namespace" not found
```

### Motivation
```
  namespaceLabelsAsTags:
    "*": prefixing_%%label%%
```
Should be enough to enable the informer

### Describe how to test/QA your changes
Refer to https://github.com/DataDog/datadog-agent/pull/25490 without step 2.

### CODE REVIEW

File: comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go [hunk 1]
```diff
@@ -47,14 +47,19 @@ func storeGenerators(cfg model.Reader) []storeGenerator {
 		generators = append(generators, newDeploymentStore)
 	}
 
+	addNamespaceStore := len(cfg.GetStringMapString("kubernetes_namespace_labels_as_tags")) > 0
```
Comment (davidor) [inline comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go:50]: We should also check `kubernetes_namespace_annotations_as_tags`.

File: comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver_test.go [hunk 2]
```diff
@@ -82,14 +82,28 @@ func TestStoreGenerators(t *testing.T) {
...
 			expectedStoresGenerator: []storeGenerator{newNodeStore, newNamespaceStore},
 		},
+		{
+			name: "Namespace from ns label as tags",
+			cfg: map[string]interface{}{
+				"kubernetes_namespace_labels_as_tags": map[string]string{
+					"env": "env",
+				},
+			},
+			expectedStoresGenerator: []storeGenerator{newNodeStore, ginewNamespaceStore},
```
Comment (jennchenn) [inline comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver_test.go:92]: was this a typo?
```suggestion
			expectedStoresGenerator: []storeGenerator{newNodeStore, newNamespaceStore},
```

File: comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go (hunk 1 above)
Comment (AliDatadog) [reply comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go]: Nice catch !

File: comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver_test.go (hunk 2 above)
Comment (AliDatadog) [reply comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver_test.go]: Indeed 👍 I just removed it. Thanks for catching it

**Review by davidor — APPROVED** [review]

**Review by jennchenn — APPROVED** [review]

**Pull request merged**